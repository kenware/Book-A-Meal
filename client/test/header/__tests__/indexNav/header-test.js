import React from 'react';
import { shallow } from 'enzyme';
import IndexNav from '../../../../components/header/index';

describe('Testing of sidebar component', () => {
  it('should render properly', () => {
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render properly with user role', () => {
    window.localStorage = {
      getItem() {
        return 'user';
      }
    };
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render properly with admin role', () => {
    window.localStorage = {
      getItem() {
        return 'admin';
      }
    };
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
<<<<<<< HEAD
    expect(tree).toMatchSnapshot();
  });

  it('should render properly with user info in the nav bar', () => {
    window.localStorage = {
      getItem() {
        return 'user';
      }
    };
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
=======
>>>>>>> bg(fix): Fixed my test and modified my controller
    expect(tree).toMatchSnapshot();
  });

  it('should respond to click method', () => {
<<<<<<< HEAD
=======
    const onClick = jest.spyOn(IndexNav.prototype, 'onClick');
>>>>>>> bg(fix): Fixed my test and modified my controller
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
    const wrapper = tree.instance();
    wrapper.onClick();
    tree.setState({ responsive: 'responsive' });
    wrapper.onClick();
    expect(tree).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('should respond to logout method', () => {
    const tree = shallow(<IndexNav onModal={jest.fn()} />);
    window.localStorage = {
      removeItem() {
      },
      getItem() {
        return 'user';
      }

    };
    global.history.pushState = jest.fn();
    const wrapper = tree.instance();
    wrapper.logout();
    expect(tree).toMatchSnapshot();
  });
});
