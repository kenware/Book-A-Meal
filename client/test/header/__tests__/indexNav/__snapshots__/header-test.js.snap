// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of sidebar component should render properly 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
    <span />
    <span />
    <span />
    <Link
      className="a login-header"
      replace={false}
      to="/login"
    >
      Login
    </Link>
    <Link
      className="a register-header"
      replace={false}
      to="/register"
    >
      SignUp
    </Link>
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should render properly with admin role 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
<<<<<<< HEAD
    <span />
    <span />
    <Link
      className="a"
      replace={false}
      to="/login"
    >
      Login
    </Link>
    <Link
      className="a"
      replace={false}
      to="/register"
    >
      SignUp
    </Link>
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should render properly with user info in the nav bar 2`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
    <span />
    <span />
    <Link
      className="a"
      replace={false}
      to="/login"
=======
    <span
      className="a logOut"
      onClick={[Function]}
      role="button"
>>>>>>> bg(fix): Fixed my test and modified my controller
    >
      Login
    </Link>
    <Link
      className="a"
      replace={false}
      to="/register"
    >
      SignUp
    </Link>
<<<<<<< HEAD
=======
    <span />
    <span />
    <span />
>>>>>>> bg(fix): Fixed my test and modified my controller
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should render properly with user role 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
<<<<<<< HEAD
    <span />
    <span />
    <Link
      className="a"
      replace={false}
      to="/login"
=======
    <span
      className="a logOut"
      onClick={[Function]}
      role="button"
>>>>>>> bg(fix): Fixed my test and modified my controller
    >
      Login
    </Link>
    <Link
      className="a"
      replace={false}
      to="/register"
    >
      SignUp
    </Link>
<<<<<<< HEAD
=======
    <span />
    <span />
    <span />
>>>>>>> bg(fix): Fixed my test and modified my controller
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should respond to click method 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
<<<<<<< HEAD
    <span />
    <span />
    <Link
      className="a"
      replace={false}
      to="/login"
=======
    <span
      className="a logOut"
      onClick={[Function]}
      role="button"
>>>>>>> bg(fix): Fixed my test and modified my controller
    >
      Login
    </Link>
    <Link
      className="a"
      replace={false}
      to="/register"
    >
<<<<<<< HEAD
      SignUp
    </Link>
=======
      <img
        alt="user"
        className="rounded-circle"
        src="admin"
        style={
          Object {
            "height": "20px",
            "width": "20px",
          }
        }
      />
    </Link>
    <span />
    <span />
    <span />
>>>>>>> bg(fix): Fixed my test and modified my controller
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;

exports[`Testing of sidebar component should respond to logout method 1`] = `
<div>
  <div
    className="nav "
    id="myTopnav"
  >
    <Link
      className="a active active-text-header"
      replace={false}
      style={
        Object {
          "fontSize": "30px",
          "marginTop": "10px",
        }
      }
      to="/meals"
    >
      Book-A-Meal
    </Link>
    <span />
<<<<<<< HEAD
    <span />
    <span />
    <Link
      className="a"
      replace={false}
      to="/login"
=======
    <span
      className="a logOut"
      onClick={[Function]}
      role="button"
>>>>>>> bg(fix): Fixed my test and modified my controller
    >
      Login
    </Link>
    <Link
      className="a"
      replace={false}
      to="/register"
    >
<<<<<<< HEAD
      SignUp
    </Link>
=======
      <img
        alt="user"
        className="rounded-circle"
        src="admin"
        style={
          Object {
            "height": "20px",
            "width": "20px",
          }
        }
      />
    </Link>
    <span />
    <span />
    <span />
>>>>>>> bg(fix): Fixed my test and modified my controller
    <span
      className="a icon"
      onClick={[Function]}
      role="button"
    >
      <span
        className="fa fa-bars"
      />
    </span>
  </div>
</div>
`;
